### DISCUSSION TITLE
Screenshot/snapshot feature only available for iOS but not Android

### DISCUSSION THREAD

**Discussion by pzyu** [google/model-viewer#1903] [Q&A]
### Description

Is the feature to take a screenshot of the placed object in AR only available for iOS Quick Look? It doesn't seem to show up on Chrome for Android.
![rover](https://user-images.githubusercontent.com/18459487/86364524-dd363180-bcaa-11ea-87b6-2bdc9bba9f3f.png)

### Browser Affected

- [X] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [X] Safari

### OS

- [X] Android
- [X] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### Versions

- model-viewer: v1.0.0

**Comment by elalish** [comment]
Indeed, we're working on bringing this feature to WebXR, but it's not available yet. However, SceneViewer (the Android equivalent of iOS QuickLook) does have this feature, so you can use `ar-modes="scene-viewer quick-look webxr"` to prioritize the modes that support screenshots.

**Comment by pzyu** [comment]
> Indeed, we're working on bringing this feature to WebXR, but it's not available yet. However, SceneViewer (the Android equivalent of iOS QuickLook) does have this feature, so you can use `ar-modes="scene-viewer quick-look webxr"` to prioritize the modes that support screenshots.

I do have ar-modes="scene-viewer quick-look webxr" included, but I do not get the button to take a screenshot, or even the title and link.

![img](https://user-images.githubusercontent.com/18459487/86380485-b2ee6f00-bcbe-11ea-96ea-ba6175b34c96.jpeg)

**Comment by elalish** [comment]
Oh, actually I should have said to remove webxr entirely from the list; I think that's where you're ending up still. I bet you hit #1286, which is already fixed and will be part of the next release.

**Comment by pzyu** [comment]
I actually realized that when I revert the model-viewer version back to 0.8.0, the screenshot feature becomes available again. I think it has to do with #1142 where the title and link have been removed. In that case, I will probably just modify the model-viewer.js to suit my needs.

**Comment by vortice3D** [comment]
Hi @elalish and @pzyu:

Correct me if I am wrong, but with last release (v1.3.0) the **screenshot functionality** (button) is still missing on **Android-Chrome**, even using `ar-modes="scene-viewer quick-look"` as stated before in this same thread.

Isn't it?

**Comment by elalish** [comment]
@vortice3D It's working for me here: https://modelviewer.dev/examples/augmentedreality/#sceneViewer

What OS version and device are you testing on? It's possible the recording feature is only available in more recent versions.

**Comment by vortice3D** [comment]
Hi @elalish:

First, thank you very much for your immediate response and, by the way, for your great job in this forum.

The "official examples" link you're addressing works also OK for me.

After double-check my code, that as usual with this technology is really simple. I'm realizing that:

**1.** **When I don´t use the ar-modes** parameter at all, it works as expected:

```
<!DOCTYPE html>
<html lang="es">
	<head>
		<title>Web Model</title>
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
		
		<style>
			body,
			html {
				height: 100%;
				margin: 0;
				padding: 0;
				width: 100%;
			}

			#model {
				background: #ff7900;
				height: 100%;
				width: 100%;
			}
		</style>
		
		<!-- Import the component -->
		<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
		<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
	</head>

	<body>
		<model-viewer id="model" alt="Astronauta" ar ar-scale="auto" camera-controls src="assets/Astronaut.glb"></model-viewer>
	</body>
</html>
```

![Screenshot_20201113-134706](https://user-images.githubusercontent.com/1760601/99075771-524f3700-25ba-11eb-8fd9-9f59d7d94a53.png)

![Screenshot_20201113-135638](https://user-images.githubusercontent.com/1760601/99075932-96dad280-25ba-11eb-9e93-a635a08b9d34.png)


**2.** However, when I add the ar-modes parameter, forcing the use of native **scene-viewer** instead of **webxr**, this way:

    ```
    <body>
         <model-viewer id="model" alt="Astronauta" ar ar-modes="scene-viewer webxr" ar-scale="auto" camera-controls src="assets/Astronaut.glb"></model-viewer>
    </body>
</html>
```

for getting so access to Record Button functionality, I can see the model in SceneViewer but not when passing to the AR experience:

![Screenshot_20201113-135814](https://user-images.githubusercontent.com/1760601/99076003-b540ce00-25ba-11eb-891e-402cf8dc18ee.png)

What translated to English says: Couldn't load object. Looks like there's something wrong with this object.

All this in the following devices:

- Motorola MOTO G6 / Android 9 / Chrome  86.0.4240.185
- Samsung Tab S4 / Android 10 / Chrome  86.0.4240.185

Thanks for your time.

**Comment by elalish** [comment]
@tpsiaki Looks like this is another glTF parsing error in Scene Viewer? I'm surprised this is happening with the astronaut model, since it's so simple. Maybe it's specific to older Android versions?

**Comment by vortice3D** [comment]
Hi there, @elalish and @tpsiaki:

Don't panic! I've just found where the problem is ... but don't understand why.

The case is that I'm hosting this playground project in a private folder of my web server. So when I launch it, a user and password must be provided. After doing so, and as I showed you in my previous post, the non-ar-experience part is shown without problems:

![Screenshot_20201116-0829a23](https://user-images.githubusercontent.com/1760601/99225027-abe87900-27e7-11eb-8d28-5fe081baedf0.jpg)


And here comes the strange behavior:

- When no "ar-modes" parameter is used, AR-experience is launched OK.

- But with "ar-modes" set, something prevents the AR experience to being launched.


This said, I'm pretty sure of that using a **private-but-already-granted-access** content is here the problem, mainly because moving all (HTML and GLTF) to a public folder of the same server, does the trick, and now I can view the astronaut in AR and with the record button (courtesy of "ar-modes"):

![Screenshot_20201116-082923](https://user-images.githubusercontent.com/1760601/99224735-2cf34080-27e7-11eb-9ce2-dfc325f05448.jpg)


Any help with this? I mean how to host a content (GLTF/GLB/UDSZ) in a private folder meanwhile can be shown in AR and with Record Button (in other words, shown without WebXR).

**Comment by elalish** [comment]
@vortice3D Ah, that does make sense. Remember that scene-viewer and quick-look (unlike webxr) are completely separate apps that share nothing with the browser. The only thing we pass to them is the model's URL, so they need to get permissions on their own. It may be possible to achieve what you want using query parameters on that URL to pass authentication, but that'll be on you to figure out.

**Comment by vortice3D** [comment]
Hi @elalish:

Yes, that is what I was thinking last night, that it would be an authentication problem of the native AR functionality. Anyway, thank you very much for all the info. I'll try with query parameters.

Anyway, when testing it on Android, keeping pressed the record button a video recording is carried out. This is not possible under iOS, that only lets you take screenshots. I suppose that this is also a Quick-look limitation. Isn't it?

Finally, about that you're saying at the beginning of this thread: "_...we're working on bringing this feature to WebXR, but it's not available yet._" Do you have any date or milestone for it?

Best regards.
